src/core/sys/linux/perf_event.d

```diff
@@ -0,0 +1,1873 @@
+        PERF_SAMPLE_BRANCH_NO_TX = 1U << perf_branch_sample_type_shift.PERF_SAMPLE_BRANCH_NO_TX_SHIFT,
+        PERF_SAMPLE_BRANCH_COND = 1U << perf_branch_sample_type_shift.PERF_SAMPLE_BRANCH_COND_SHIFT,
+
+        PERF_SAMPLE_BRANCH_CALL_STACK = 1U << perf_branch_sample_type_shift.PERF_SAMPLE_BRANCH_CALL_STACK_SHIFT,
+        PERF_SAMPLE_BRANCH_IND_JUMP = 1U << perf_branch_sample_type_shift.PERF_SAMPLE_BRANCH_IND_JUMP_SHIFT,
+        PERF_SAMPLE_BRANCH_CALL = 1U << perf_branch_sample_type_shift.PERF_SAMPLE_BRANCH_CALL_SHIFT,
+
+        PERF_SAMPLE_BRANCH_NO_FLAGS = 1U << perf_branch_sample_type_shift.PERF_SAMPLE_BRANCH_NO_FLAGS_SHIFT,
+        PERF_SAMPLE_BRANCH_NO_CYCLES = 1U << perf_branch_sample_type_shift.PERF_SAMPLE_BRANCH_NO_CYCLES_SHIFT,
+
+        PERF_SAMPLE_BRANCH_TYPE_SAVE = 1U << perf_branch_sample_type_shift.PERF_SAMPLE_BRANCH_TYPE_SAVE_SHIFT,
+
+        PERF_SAMPLE_BRANCH_MAX = 1U << perf_branch_sample_type_shift.PERF_SAMPLE_BRANCH_MAX_SHIFT
+}
+
+enum
+{
+    PERF_BR_UNKNOWN = 0,
+    PERF_BR_COND = 1,
+    PERF_BR_UNCOND = 2,
+    PERF_BR_IND = 3,
+    PERF_BR_CALL = 4,
+    PERF_BR_IND_CALL = 5,
+    PERF_BR_RET = 6,
+    PERF_BR_SYSCALL = 7,
+    PERF_BR_SYSRET = 8,
+    PERF_BR_COND_CALL = 9,
+    PERF_BR_COND_RET = 10,
+    PERF_BR_MAX = 11
+}
+
+enum PERF_SAMPLE_BRANCH_PLM_ALL = perf_branch_sample_type.PERF_SAMPLE_BRANCH_USER
+    | perf_branch_sample_type.PERF_SAMPLE_BRANCH_KERNEL
+    | perf_branch_sample_type.PERF_SAMPLE_BRANCH_HV;
+
+enum perf_sample_regs_abi
+{
+    PERF_SAMPLE_REGS_ABI_NONE = 0,
+    PERF_SAMPLE_REGS_ABI_32 = 1,
+    PERF_SAMPLE_REGS_ABI_64 = 2
+}
+
+enum
+{
+    PERF_TXN_ELISION = 1 << 0,
+    PERF_TXN_TRANSACTION = 1 << 1,
+    PERF_TXN_SYNC = 1 << 2,
+    PERF_TXN_ASYNC = 1 << 3,
+    PERF_TXN_RETRY = 1 << 4,
+    PERF_TXN_CONFLICT = 1 << 5,
+    PERF_TXN_CAPACITY_WRITE = 1 << 6,
+    PERF_TXN_CAPACITY_READ = 1 << 7,
+
+    PERF_TXN_MAX = 1 << 8,
+
+    PERF_TXN_ABORT_SHIFT = 32
+}
+enum perf_event_read_format
+{
+    PERF_FORMAT_TOTAL_TIME_ENABLED = 1U << 0,
+    PERF_FORMAT_TOTAL_TIME_RUNNING = 1U << 1,
+    PERF_FORMAT_ID = 1U << 2,
+    PERF_FORMAT_GROUP = 1U << 3,
+
+    PERF_FORMAT_MAX = 1U << 4
+}
+
+enum PERF_ATTR_SIZE_VER0 = 64;
+enum PERF_ATTR_SIZE_VER1 = 72;
+enum PERF_ATTR_SIZE_VER2 = 80;
+enum PERF_ATTR_SIZE_VER3 = 96;
+
+enum PERF_ATTR_SIZE_VER4 = 104;
+enum PERF_ATTR_SIZE_VER5 = 112;
+
+struct perf_event_attr
+{
+
+    uint type;
+
+    uint size;
+
+    ulong config;
+
+    union
+    {
+        ulong sample_period;
+        ulong sample_freq;
+    }
+
+    ulong sample_type;
+    ulong read_format;
+    private ulong _disabled_inherit_pinned_exclusive_exclude_user_exclude_kernel_exclude_hv_exclude_idle_mmap_comm_freq_inherit_stat_enable_on_exec_task_watermark_precise_ip_mmap_data_sample_id_all_exclude_host_exclude_guest_exclude_callchain_kernel_exclude_callchain_user_mmap2_comm_exec_use_clockid_context_switch_write_backward_namespaces___reserved_1;
```

Comment:
seriously ? It is private so the mangling does not matter. Reduce the size of this identifier, please.

Reply:
eventually you can keep the long stuff in a comment, as it is certainly a very very self documenting name.